go/libraries/doltcore/sqle/database.go

```diff
@@ -140,6 +153,41 @@ func NewDatabase(name string, dbData env.DbData, editOpts editor.Options) Databa
 	}
 }
 
+// GetInitialDBState returns the InitialDbState for |db|.
+func GetInitialDBState(ctx context.Context, db Database) (dsess.InitialDbState, error) {
```

Comment:
Add a note here that this is only to support the creation of in-memory databases.

The method of using repo state to initialize these things is pretty janky